source/slang/slang-ir-legalize-varying-params.cpp

```diff
@@ -1104,7 +1104,7 @@ struct CUDAEntryPointVaryingParamLegalizeContext : EntryPointVaryingParamLegaliz
     {
         for (auto attr : typeLayout->getSizeAttrs())
         {
-            if (attr->getSize() != 0)
+            if (attr->getSize().compare(0) == std::partial_ordering::greater)
```

Comment:
Do the IR layout instruction accessors not return the `LayoutSize` and `LayoutOffset` types? If not, that seems like something we should fix over time. Having them return raw integer values seems dangerous, and this comparison seems to show exactly why that is.